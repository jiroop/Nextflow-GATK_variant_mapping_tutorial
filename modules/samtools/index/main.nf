// Generate BAM index file
process SAMTOOLS_INDEX {

    container 'quay.io/biocontainers/samtools:1.20--h50ea8bc_1'

    publishDir params.outdir, mode: 'symlink'

    input:
        path input_bam

    output:
        tuple path(input_bam), path("${input_bam}.bai") 

    script:
    """
    samtools index "${input_bam}"
    """

}