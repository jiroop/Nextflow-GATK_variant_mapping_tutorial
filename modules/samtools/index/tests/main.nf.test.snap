{
    "Should index reads_son.bam correctly": {
        "content": [
            {
                "0": [
                    [
                        "reads_son.bam:md5,af5956d9388ba017944bef276b71d809",
                        "reads_son.bam.bai:md5,a2ca7b84998218ee77eff14af8eb8ca2"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-25T15:08:27.170122"
    },
    "Should index reads_father.bam correctly": {
        "content": [
            {
                "0": [
                    [
                        "reads_father.bam:md5,57f3b977b0eee8278924e38894d02a57",
                        "reads_father.bam.bai:md5,49cb8454781a279a8b33f47209e76fe5"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-25T15:12:23.045177"
    },
    "Should index reads_mother.bam correctly": {
        "content": [
            {
                "0": [
                    [
                        "reads_mother.bam:md5,04b918c3707c24d8f3a7e88bb68e2d32",
                        "reads_mother.bam.bai:md5,1b17cbcc9392e7e787cc83a9081deb31"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-25T15:12:24.949021"
    }
}